scipy/stats/tests/test_stats.py

```diff
@@ -7971,3 +7972,124 @@ def test_rename_mode_method(fun, args):
     with pytest.raises(TypeError, match=re.escape(err)):
         fun(*args, method='exact', mode='exact')
 
+
+class TestExpectile:
+    def test_same_as_mean(self):
+        rng = np.random.default_rng(42)
+        x = rng.random(size=20)
+        assert_allclose(stats.expectile(x, alpha=0.5), np.mean(x))
+
+
+    def test_minimum(self):
+        rng = np.random.default_rng(42)
+        x = rng.random(size=20)
+        assert_allclose(stats.expectile(x, alpha=0), np.amin(x))
+
+    def test_maximum(self):
+        rng = np.random.default_rng(42)
+        x = rng.random(size=20)
+        assert_allclose(stats.expectile(x, alpha=1), np.amax(x))
+   
+    @pytest.mark.parametrize("alpha", [0.2, 0.5, 0.8])
+    @pytest.mark.parametrize("n", [20, 2000])
+    def test_expectile_properties(self, alpha, n):
+        """
+        See Section 6 of
+        I. Steinwart, C. Pasin, R.C. Williamson & S. Zhang (2014).
+        "Elicitation and Identification of Properties". COLT.
+        http://proceedings.mlr.press/v35/steinwart14.html
+
+        and
+
+        Propositions 5, 6, 7 of
+        F. Bellini, B. Klar, and A. Müller and E. Rosazza Gianin (2013).
+        "Generalized Quantiles as Risk Measures"
+        http://doi.org/10.2139/ssrn.2225751 
+        """
+        rng = np.random.default_rng(42)
+        x = rng.normal(size=n)
+
+        # 0. definite / constancy
+        # Let T(X) denote the expectile of rv X ~ F.
+        # T(c) = c for constant c
+        for c in [-5, 0, 0.5]:
+            assert_allclose(
+                stats.expectile(np.full(shape=n, fill_value=c), alpha=alpha),
+                c
+            )
+
+        # 1. translation equivariance
+        # T(X + c) = T(X) + c
+        c = rng.exponential()
+        assert_allclose(
+            stats.expectile(x + c, alpha=alpha),
+            stats.expectile(x, alpha=alpha) + c,
+        )
+        assert_allclose(
+            stats.expectile(x - c, alpha=alpha),
+            stats.expectile(x, alpha=alpha) - c,
+        )
+
+        # 2. positively homogeneity
+        # T(cX) = c * T(X) for c > 0
+        assert_allclose(
+            stats.expectile(c * x, alpha=alpha),
+            c * stats.expectile(x, alpha=alpha),
+        )
+
+        # 3. subadditivity
+        # Note that subadditivity holds for alpha >= 0.5.
+        # T(X + Y) <= T(X) + T(Y)
+        # For alpha = 0.5, i.e. the mean, strict euqality holds.
+        # For alpha < 0.5, one can use property 6. to show
+        # T(X + Y) >= T(X) + T(Y)
+        y = rng.logistic(size=n, loc=10)  # different distibution than x
+        if alpha == 0.5:
+            def op(a, b):
+                return a == pytest.approx(b)
+
+        else:
+            op = operator.le if alpha > 0.5 else operator.gt
+        assert op(
+            stats.expectile(np.r_[x + y], alpha=alpha),
+            stats.expectile(x, alpha=alpha)
+            + stats.expectile(y, alpha=alpha)
+        )
```

Comment:
Thanks for adapting this.

If we want to test `alpha=0.5` from both sides, we could also do something as follows (sorry for the back-and-forth).

```suggestion
        if alpha >= 0.5:
            # subadditiviy
            assert (
                stats.expectile(np.r_[x + y], alpha=alpha)
                <= stats.expectile(x, alpha=alpha)
                + stats.expectile(y, alpha=alpha)
            )
       if alpha <= 0.5:
            # superadditivity
            assert (
                stats.expectile(np.r_[x + y], alpha=alpha)
                >= stats.expectile(x, alpha=alpha)
                + stats.expectile(y, alpha=alpha)
            )
```

Reply:
I prefer the current version with `op`, because alpha=0.5 needs special care (due to numerical precision) and because it is reused later.

Reply:
> due to numerical precision

Does this mean that the inequalities for alpha close to 0.5 would need some tolerances as well? If so, that'd would at least be worth a comment IMO

Reply:
It does not seem to be the case. I added alpha=0.5 + 1e-12.